src/oc_erchef/apps/chef_objects/src/chef_cbv_cache.erl

```diff
@@ -0,0 +1,253 @@
+         claim/1,
+         put/2,
+         %% Debug and testing
+         force_breaker/1,
+         stop/0
+        ]).
+
+%% gen_server callbacks
+-export([
+         code_change/3,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         init/1,
+         terminate/2
+        ]).
+
+-spec start_link() -> {ok, pid()} | ignore | {error, term()}.
+start_link() ->
+    Enabled = envy:get(chef_objects, cbv_cache_enabled, false, boolean),
+    TTL = envy:get(chef_objects, cbv_cache_item_ttl, ?DEFAULT_TTL, integer),
+    gen_server:start_link({local, ?SERVER}, ?MODULE, [Enabled, TTL], []).
+
+%% @doc this function retrieves a stored value from the cache.
+%% If the value is missing, it will return undefined.
+%% If another caller has claimed intent to populate the key, this will return
+%% 'retry' indicating that the caller should try again after a brief delay.
+%% If it returns 'busy' the process is not available to service the request, and
+%% the caller should fail without retrying.
+-spec get(any()) -> retry | busy | undefined | term().
+get(Key) ->
+    send_if_available({get, Key}).
+
+%% @doc claim intent to populate Key into the cache.
+%% When the claim is accepted, returns 'ok'.
+%% When another process has already claimed intent, this returns 'retry' indicating
+%% that the caller should retry the original 'get' request after a brief delay.
+%% If it returns 'busy' the process is not available to service the request, and
+%% the caller should fail without retrying.
+-spec claim(any()) -> ok | retry | busy.
+claim(Key) ->
+    send_if_available({claim, Key}).
+
+%% @doc store a value in the cache.  Caller must have first claimed intent to populate
+%% this key by invoking claim/1.
+%% If the caller has not first invoked claim/1, this will return {error, noclaim}.
+%% If another process has invoked claim/1, this will return {error, already_claimed}.
+%% returns 'ok' when the value is stored successfully
+-spec put(any(), term()) -> {error, already_claimed} | {error, noclaim} | ok.
+put(Key, Value) ->
+    % We do not protect this with the breaker - this function is only to be called
+    % after a successful 'claim', and we want to allow caller to put the claim
+    % instead of `busy` response which would cause another caller to have to claim,
+    % do the work, then put the result.
+    gen_server:call(?SERVER, {put, Key, Value}).
+
+%% Internal function that checks in with the process message queue for chef_cbv_cache intermittently
+%% and tracks whether its mailbox has grown to large. If that happens, response with 'busy' instead
+%% of sending the request through.
+send_if_available(Msg) ->
+    case breaker_tripped() of
+        true -> busy;
+        false -> gen_server:call(?SERVER, Msg)
+    end.
+
+%% @doc intended for testing.
+stop() ->
+    gen_server:call(?SERVER, stop).
+
+% gen_server implementation
+
+init([Enabled, TTL]) ->
+    process_flag(trap_exit, true),
+    spawn_breaker(),
```

Comment:
Would it be possible to enable spawn_breaker()/breaker() process only if 'cbv_cache_enabled = true' as 'erlang:send_after(?QUEUE_LEN_REFRESH_INTERVAL, self(), update)' makes this process busy unnecessarily if 'cbv_cache_enabled = false'.